### PR TITLE
Framework: Replace flowRight occurences with compose for HOCs

### PR DESCRIPTION
**Pull request by gziolo**
## Description

In #3493 new utility method `compose` was introduced to be used with Higher-Order Components. This PR refactors all occurrences of `flowRight` around HOCs to be replaced with `compose`.

## How Has This Been Tested?

Manually. All tests should still pass, linter should be happy and editor should work without errors.

## Types of changes

Refactoring only. No changes in the existing functionalities.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by youknowriad** [comment]
What about `flow` usage, I think we have some as well :)

**Comment by gziolo** [comment]
So I have just learned that `flow` and `flowRight` were used as if they were the same. That makes replacing `flow` easier and validates the need of `compose` helper method :)

**Comment by aduth** [comment]
>So I have just learned that `flow` and `flowRight` were used as if they were the same.

That's not entirely true; in many cases we're assuming props to be passed from a previous HoC in the array, which works with `flowRight`, but not `flow`. See also: https://github.com/WordPress/gutenberg/pull/2500#discussion_r134669757

**Comment by gziolo** [comment]
@aduth if you will take a closer look, you will notice that `connect` almost always goes first and it doesn't matter whether it's used with `flow` or `flowRight`. This is where my comment comes from. If you are aware of any place where I should update the order, let me know.

File: editor/components/inserter/menu.js [hunk 1]
```diff
@@ -365,7 +365,7 @@ const connectComponent = connect(
 	{ showInsertionPoint, hideInsertionPoint, fetchReusableBlocks }
 );
 
-export default flow(
+export default compose(
```
Comment (gziolo) [inline editor/components/inserter/menu.js]: @aduth It might be an issue only here. I will flip the order to be safe.

Comment (gziolo) [reply editor/components/inserter/menu.js]: Order reversed 👍

**Review by aduth — APPROVED** [review]

**Pull request merged**